### PR TITLE
Update idler - fixes ignoring dc events

### PR DESCRIPTION
**Pull request by sthaha**
This update includes the following changes
  -  Improvements to Idle/UnIdle condition evaluation
  -  Add more logging to openshift-client
  -  Reverts "Remove DC and user condition (#285)"
  -  Reverts "Remove unused WatchDeployment in openshift client"

### CODE REVIEW

File: dsaas-services/f8-jenkins-idler.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 services:
-- hash: 0292a87c27918b8010abd1586abf04bc93f6febf
+- hash: f2054459a5209d04723fc3b2b3f022acad0b4228
```
Comment (sthaha) [inline dsaas-services/f8-jenkins-idler.yaml]: https://github.com/fabric8-services/fabric8-jenkins-idler/commit/f2054459a5209d04723fc3b2b3f022acad0b4228

**Comment by chmouel** [comment]
@ppitonak @ljelinkova Those are failing due of the idler issue it seems, since this commit is here to (hopefully 🤞🏼) fix this, can we force merge ?

**Comment by ppitonak** [comment]
@chmouel these fixed didn't help prod-preview much :( you probably included wrong issue to PR description.

**Comment by sthaha** [comment]
> @chmouel these fixed didn't help prod-preview much :( you probably included wrong issue to PR description.

@ppitonak what do you see on prod-preview ? Have you tried running the test against your account?

**Comment by ppitonak** [comment]
@sthaha Jenkins was not able to unidle on my two accounts when I wrote the comment. It did unidle today.

**Pull request merged**